### PR TITLE
fix execute when init in the container already exists

### PR DESCRIPTION
**Pull request by tych0**
see patch notes for details

### CODE REVIEW

File: src/lxc/execute.c [hunk 1]
```diff
@@ -58,11 +58,15 @@ static int execute_start(struct lxc_handler *handler, void* data)
...
 	argv = malloc((argc + argc_add) * sizeof(*argv));
-	if (!argv)
+	if (!argv) {
+		SYSERROR("Allocating init args failed");
 		goto out1;
+	}
 
-	if (!my_args->init_path)
+	if (!my_args->init_path) {
+		ERROR("Init path missing?");
```
Comment (hallyn) [inline src/lxc/execute.c]: Why the question mark?

File: src/lxc/conf.c [hunk 2]
```diff
@@ -3213,8 +3213,12 @@ static int lxc_execute_bind_init(struct lxc_handler *handler)
 	/* If init exists in the container, don't bind mount a static one */
 	p = choose_init(conf->rootfs.mount);
 	if (p) {
-		free(p);
-		return 0;
+		char *old = p;
+
+		p = strdup(old + strlen(conf->rootfs.mount));
```
Comment (hallyn) [inline src/lxc/conf.c]: Check for alloc failure (before printing in INFO())

**Comment by hallyn** [comment]
Two little comments, otherwise looks good to me - thanks.

File: src/lxc/execute.c (hunk 1 above)
Comment (tych0) [reply src/lxc/execute.c]: Heh, no reason :)

**Comment by tych0** [comment]
Fixed, thanks. Derp'd the allocation failure there :)

File: src/lxc/conf.c [hunk 3]
```diff
@@ -3213,8 +3213,15 @@ static int lxc_execute_bind_init(struct lxc_handler *handler)
...
+		if (!p)
```
Comment (hallyn) [inline src/lxc/conf.c]: But you need to free(old) in this case right?

Comment (tych0) [reply src/lxc/conf.c]: yes, thanks

**Comment by hallyn** [comment]
@tych0 sorry - by merging @brauner 's commit i caused a trivial conflict in yours.  Can you rebase?

**Comment by tych0** [comment]
Done

**Pull request merged**